src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,116 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Status } from '../../types/Status';
 
 export const TodoList: React.FC = () => {
+  const list = useAppSelector(state => state.todos);
+  const [todos, setTodos] = useState(list);
+  const currentTodoId = useAppSelector(state => state.currentTodo?.id);
+  const { query, status } = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+  const setTodoActive = (todo: Todo) => dispatch(actions.setTodo(todo));
+
+  const filterData = (data: Todo[], search: string, searchStatus: Status) => {
+    if (search === '' && searchStatus === 'all') {
+      return data;
+    }
+
+    return data.filter(el => {
+      switch (searchStatus) {
+        case 'active':
+          return !el.completed;
+        case 'completed':
+          return el.completed;
+        case 'all':
+        default:
+          return el;
+      }
+    }).filter(todo => (
```

Comment:
You're filtering the data twice. It's better to use one filter and perform all the filtering logic inside it to improve performance.